### ISSUE TITLE
Implement multipart file uploads

### ISSUE THREAD

**Issue by abonander** [rustasync/surf#75]
I couldn't find anywhere documenting a desire or otherwise a lack of desire for `multipart/form-data` support.

I have a very basic streaming multipart client API in `multipart-async` now: https://github.com/abonander/multipart-async/tree/master/src/client

It's not quite production ready yet though. I want to add several more tests. It's also built against `tokio-io` instead of `futures-io`.

**Comment by yoshuawuyts** [comment]
@abonander yeah, having this would be great! -- I considered implementing it before the initial release, but this *is* somewhat tricky and we'd need to come up with a good design first. Would you like to work on bringing this into Tide?

**Comment by abonander** [comment]
@yoshuawuyts Tide or Surf? I also have a server API that's just about ready for use which would close https://github.com/rustasync/tide/issues/89

**Comment by futurist** [comment]
What's the plan of this feature? It's an important feature I think (or bug?)

**Comment by Fishrock123** [comment]
See https://github.com/http-rs/http-types/pull/175 - likely to picked back up soon over there, unsure how or when that will impact a full surf api for this.

**Comment by Fishrock123** [comment]
I don't have any investment in this anymore. Someone else will need to pick it up.
